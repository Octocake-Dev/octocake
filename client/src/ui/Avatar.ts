import Image from "next/image";

import { styled } from "../../stitches.config";

const Avatar = styled(Image, {
  borderRadius: "10px",
  backgroundColor: "$primary300",
});

export default Avatar;
