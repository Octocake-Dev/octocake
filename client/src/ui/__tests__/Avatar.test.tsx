import React from "react";

import { render, screen } from "@testing-library/react";

import Avatar from "@/ui/Avatar";

describe("Avatar", () => {
  const alt = "avatar";

  it("Should render an avatar", () => {
    render(<Avatar src="/logo.svg" width="50" height="50" alt={alt} />);

    expect(screen.getByAltText(alt)).toBeVisible();
  });
});
