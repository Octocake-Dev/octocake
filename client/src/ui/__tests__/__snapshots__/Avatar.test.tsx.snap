// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Should match snapshot 1`] = `
<img
  alt="avatar"
  class="c-hutZiQ"
  data-nimg="intrinsic"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
/>
`;
